src/app_manager/app.py

```diff
@@ -153,12 +153,12 @@ def load_Interface_from_file(filename):
     """
     with open(filename,'r') as f:
         y = yaml.load(f.read())
-        y = y or {} #coerce to dict
-        try:
-            subscribed_topics = y.get('subscribed_topics', {})
-            published_topics = y.get('published_topics', {})
-        except KeyError:
-            raise InvalidAppException("Malformed interface, missing keys")
+    y = y or {} #coerce to dict
+    try:
+        subscribed_topics = y['subscribed_topics'] if 'subscribed_topics' in y else {}
```

Comment:
I don't see how converting all these places to a more verbose syntax helps with the speedup.
At least with my local profiling there is literally no difference between `a.get(b,'')` and `a[b] if b in a else ''`.
What's your benchmarking result compared to leaving all these replacements out?

Reply:
In my env, it got faster a little bit less than 1 seconds.
But it is really a small difference.

Reply:
with `dict.get`

```
[ERROR] [1640619851.553788]: total time: 4.12315011024
```

no `dict.get`

```
[ERROR] [1640619765.145302]: total time: 3.93386387825
```

Reply:
However, `dict.get` have not big change, so I will remove the change.

Reply:
part of that might be noise, I wouldn't even expect 0.2s difference based on my minor tests here. But you are right, even 0.2s is not enough to justify these changes :+1: